freqtrade/optimize/backtesting.py

```diff
@@ -312,7 +314,24 @@ def _enter_trade(self, pair: str, row: List) -> Optional[LocalTrade]:
             stake_amount = self.wallets.get_trade_stake_amount(pair, None)
         except DependencyException:
             return None
-        min_stake_amount = self.exchange.get_min_pair_stake_amount(pair, row[OPEN_IDX], -0.05)
+
+        stake_amount = strategy_safe_wrapper(self.strategy.custom_stake_amount,
+                                             default_retval=None)(
+            pair=pair, current_time=row[DATE_IDX].to_pydatetime(), current_rate=row[OPEN_IDX],
+            proposed_stake=stake_amount)
+
+        min_stake_amount = self.exchange.get_min_pair_stake_amount(pair, row[OPEN_IDX], -0.05) or 0
+        max_stake_amount = self.wallets.get_available_stake_amount()
+        if min_stake_amount is not None and stake_amount < min_stake_amount:
+            logger.error(
+                f"Stake amount for pair {pair} is too small ({stake_amount} < {min_stake_amount})"
+            )
+            return None
+        if stake_amount > max_stake_amount:
+            logger.error(
+                f"Stake amount for pair {pair} is too big ({stake_amount} > {max_stake_amount})"
+            )
+            return None
```

Comment:
Is there a particular reason to reimplement this validation here instead of using `_validate_stake_amount` (maybe moved to a different location)?

Reply:
Ah this was stupid mypy complaining about "too complex method". I would have kept it in original place..